### PR TITLE
add verified parameter to url

### PR DESCRIPTION
**Pull request by linzichao**
Issue https://github.com/Homebrew/homebrew-cask/issues/79028

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -270,6 +271,13 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def check_url_verification
+      return unless cask.url.verified
+      return if cask.url.to_s.start_with?(cask.url.verified)
+
+      add_error "url #{cask.url} does not start with #{cask.url.verified}"
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/audit.rb:278]: It would be useful to add a [link to the documentation](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment), so the user knows how to proceed.

Technically the documentation URL will change once this is is accepted, but it’s better to have something in place now.

Also, since this replaces the `style` check, it should be removed in this PR as well (fine if it’s another commit).

File: Library/Homebrew/cask/audit.rb [hunk 2]
```diff
@@ -270,6 +271,13 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def check_url_verification
+      return unless cask.url.verified
+      return if cask.url.to_s.start_with?(cask.url.verified)
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/audit.rb:276]: The protocol (`http[s]://`) shouldn’t be required for verification.

File: Library/Homebrew/cask/audit.rb (hunk 1 above)
Comment (vitorgalvao) [reply Library/Homebrew/cask/audit.rb:278]: You can use https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-verified as the documentation link. I’ve made a [documentation PR](https://github.com/Homebrew/homebrew-cask/pull/84607) for when this is one is ready.

File: Library/Homebrew/cask/audit.rb (hunk 2 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: I use regular expression to remove `http[s]://` prefix.

File: Library/Homebrew/cask/audit.rb [hunk 3]
```diff
@@ -270,6 +271,16 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def check_url_verification
+      return unless cask.url.verified
+
+      url = cask.url.to_s.sub(%r{^(http|https)://}, "")
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/audit.rb:277]: Simpler regex: `^https?://`.

**Comment by vitorgalvao** [comment]
> Also, since this replaces the style check, it should be removed in this PR as well (fine if it’s another commit).

That one is still missing, but I think it should be good to go once that’s done.

File: Library/Homebrew/cask/audit.rb [hunk 4]
```diff
@@ -270,6 +271,16 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def check_url_verification
+      return unless cask.url.verified
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/audit.rb:275]: This will make it succeed if there is no `verified:` parameter, but that’s not the goal of the check. The point is to detect *when it’s needed but it’s not present*, and if it’s correct when present.

It should:

* Check if `url` and and `homepage` have the same hostname.
* If they do, we can `return`.
* If they don’t, check if `url` and `verified` match.
* If they do, we can `return`.
* If they don’t, throw the error and point to the documentation.

It might be useful to consult the `style` check and see how it’s done there. That check should also be removed in this PR.

In addition, it would be helpful if some common hosts had longe checks. For example, a Github repo download follows the form `https://github.com/user/repo`. In those cases, we should check that much (not just `github.com`) before marking as correct. That said, that can be done afterwards if you don’t want to pursue it that far. But for the moment this PR should make the `audit` at least as good as the `style` check.

File: Library/Homebrew/cask/url.rb [hunk 5]
```diff
@@ -5,7 +5,7 @@ class URL
     :using,
     :tag, :branch, :revisions, :revision,
     :trust_cert, :cookies, :referer, :user_agent,
-    :data
+    :data, :verified
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/url.rb:8]: Make it the first one. This will be the most used parameter.

File: Library/Homebrew/cask/audit.rb [hunk 6]
```diff
@@ -270,6 +271,16 @@ def bad_osdn_url?
...
+
+      url = cask.url.to_s.sub(%r{^(http|https)://}, "")
+      verification = cask.url.verified.sub(%r{^(http|https)://}, "")
+      return if url.start_with?(verification)
+
+      add_error "URL verification failed. See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-verified"
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/audit.rb:281]: `URL verification failed` is too generic. It should roughly explain what that means.

**Review by vitorgalvao — DISMISSED** [review]

**Comment by linzichao** [comment]
@vitorgalvao, thanks for your detail direction!
This time I mainly focus on migrating `style` check to `audit` from [homepage_matches_url.rb](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/cask/homepage_matches_url.rb).

**Comment by vitorgalvao** [comment]
It seems like a lot of unnecessary small functions, but I understand you just moved the logic over, which is fine as a first pass. Thank you! @reitermarkus You were the one to make the logic for `style`, so care to take a look here and see if anything should be different?

File: Library/Homebrew/cask/audit.rb [hunk 7]
```diff
@@ -270,6 +273,83 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def extract_url(url)
+      url.to_s.gsub(%r{.*"([a-z0-9]+://[^"]+)".*}m, '\1')
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:277]: I don't think this method is necessary since we don't store the URL with quotes.

File: Library/Homebrew/cask/audit.rb [hunk 8]
```diff
@@ -270,6 +273,83 @@ def bad_osdn_url?
...
+    end
+
+    def remove_non_ascii(string)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:280]: Why is this necessary?

File: Library/Homebrew/cask/audit.rb [hunk 9]
```diff
@@ -270,6 +273,83 @@ def bad_osdn_url?
...
+      string.gsub(/\P{ASCII}/, "")
+    end
+
+    def homepage
+      URI(remove_non_ascii(extract_url(cask.homepage))).host
+    end
+
+    def domain
+      strip_url_scheme(extract_url(cask.url)).gsub(%r{^([^/]+).*}, '\1')
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:289]: I think this can just be `URI(cask.url.to_s).host`.

File: Library/Homebrew/cask/audit.rb (hunk 7 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: Sure.

File: Library/Homebrew/cask/audit.rb (hunk 8 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: I just follow the logic from `style`. I remove it.

File: Library/Homebrew/cask/audit.rb (hunk 9 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: Sure

**Comment by vitorgalvao** [comment]
@reitermarkus Is anything missing from your requests, or is this good to go?

**Comment by MikeMcQuaid** [comment]
@reitermarkus nudge on thoughts on this.

**Comment by reitermarkus** [comment]
Looking at this again, I'm not sure moving this into `audit` is better than keeping this as a RuboCop.

**Comment by vitorgalvao** [comment]
> I'm not sure moving this into `audit` is better than keeping this as a RuboCop.

I’d say it makes more sense as an `audit` because it’s a security check, it doesn’t concern itself with the style of the cask. In addition, `style` is broken for users way more often.

That said, I don’t mind whatever check it is, but I do think we shouldn’t be relying on a carefully worded comment as a feature, let alone an important one. Comments should be comments, not code.

Finally, it would improve consistency. This is the only feature we have that depends on a comment.

**Comment by core-code** [comment]
> Comments should be comments, not code.

+1

**Comment by reitermarkus** [comment]
> because it’s a security check, it doesn’t concern itself with the style of the cask

There are RuboCops that solely check for security risks.

> Comments should be comments, not code.

I don't see why a comment in this case is not as informative as a `verified` parameter. Also, the `verified` parameter could still be checked by a RuboCop.

**Comment by reitermarkus** [comment]
In any case, this needs a rebase and the RuboCop tests need to be ported to audit tests if we indeed want it to be an audit.

**Comment by linzichao** [comment]
@reitermarkus Do you have any ideas about how to port the RuboCop tests to audit tests?

**Comment by reitermarkus** [comment]
Take a look at the existing tests in `test/cask/audit_spec.rb` and `test/rubocops/cask/homepage_matches_url_spec.rb` and then test the same things with the same patterns as in `audit_spec.rb`.

**Comment by vitorgalvao** [comment]
@linzichao Are you in need of specific help or just haven’t found the time yet?

**Comment by linzichao** [comment]
@vitorgalvao Sorry, I am a little busy these days so cannot work on it.

**Comment by linzichao** [comment]
@vitorgalvao Would you mind that I rewrite the whole code? Because the past commits would conflict with current branch. It seems pretty uneasy to resolve it.

**Comment by vitorgalvao** [comment]
> Would you mind that I rewrite the whole codes? Because the past commits would conflict with the current branch. It seems pretty uneasy to resolve it.

Sure. Whatever works best for you.

**Comment by linzichao** [comment]
@vitorgalvao I resolve the conflicts. Please take a look.

File: Library/Homebrew/cask/audit.rb [hunk 10]
```diff
@@ -342,6 +345,75 @@ def bad_osdn_url?
...
+      end
+      host == home
+    end
+
+    def strip_url_scheme(url)
+      url.sub(%r{^.*://(www\.)?}, "")
+    end
+
+    def url_from_verified
+      cask.url.verified.sub(/[^ ]*# ([^ ]+) .*/, '\1').sub(%r{^https?://}, "")
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:379]: The first `.sub` is unnecessary now.

File: Library/Homebrew/cask/audit.rb [hunk 11]
```diff
@@ -342,6 +345,75 @@ def bad_osdn_url?
...
+      return unless verified_matches_url?
+
+      add_error "The URL's #{domain} matches the homepage #{homepage}, " \
+                          "the `verified` parameter of the `url` stanza is unnecessary. " \
+                                  "See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-verified"
+    end
+
+    def check_missing_verified
+      return if url_match_homepage?
+      return if !url_match_homepage? && verified?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:402]: `!url_match_homepage?` is always true here because of the first line.

File: Library/Homebrew/cask/audit.rb [hunk 12]
```diff
@@ -342,6 +345,75 @@ def bad_osdn_url?
...
+
+    def url_from_verified
+      cask.url.verified.sub(/[^ ]*# ([^ ]+) .*/, '\1').sub(%r{^https?://}, "")
+    end
+
+    def verified_matches_url?
+      strip_url_scheme(cask.url.to_s).include?(url_from_verified)
+    end
+
+    def verified?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:386]: Naming this `verified?` makes it sound like it is actually verified correctly instead of `verified:` only being present.

File: Library/Homebrew/test/cask/audit_spec.rb [hunk 13]
```diff
@@ -895,5 +895,83 @@ def tmp_cask(name, text)
         expect(subject).to pass
       end
     end
+
+    context "when the url matches the homepage and there is no comment" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/audit_spec.rb:899]: Not a comment anymore.

File: Library/Homebrew/test/cask/audit_spec.rb [hunk 14]
```diff
@@ -895,5 +895,83 @@ def tmp_cask(name, text)
...
+    end
+
+    context "when the url does not match the homepage" do
+      let(:cask_token) { "foo" }
+      let(:cask) do
+        tmp_cask cask_token.to_s, <<~RUBY
+          cask '#{cask_token}' do
+            version '1.8.0_72,8.13.0.5'
+            sha256 '8dd95daa037ac02455435446ec7bc737b34567afe9156af7d20b2a83805c1d8a'
+            # brew.sh was verified as official
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/audit_spec.rb:946]: This should not be present anymore anywhere.

File: Library/Homebrew/test/cask/audit_spec.rb [hunk 15]
```diff
@@ -895,5 +895,83 @@ def tmp_cask(name, text)
...
+            homepage 'https://foo.brew.sh'
+            app 'Audit.app'
+          end
+        RUBY
+      end
+
+      it { is_expected.to pass }
+    end
+
+    context "when the url matches the homepage but there is a comment " \
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/audit_spec.rb:918]: This doesn't test anything right now without an actual `verified` parameter being present.

File: Library/Homebrew/cask/audit.rb [hunk 16]
```diff
@@ -342,6 +345,75 @@ def bad_osdn_url?
...
+    def strip_url_scheme(url)
+      url.sub(%r{^.*://(www\.)?}, "")
+    end
+
+    def url_from_verified
+      cask.url.verified.sub(/[^ ]*# ([^ ]+) .*/, '\1').sub(%r{^https?://}, "")
+    end
+
+    def verified_matches_url?
+      strip_url_scheme(cask.url.to_s).include?(url_from_verified)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:383]: I think this can be `==` now instead of `include?`.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Library/Homebrew/cask/audit.rb (hunk 11 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb:402]: Remove it.

File: Library/Homebrew/cask/audit.rb (hunk 12 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb:386]: Rename as `verified_present?`.

File: Library/Homebrew/test/cask/audit_spec.rb (hunk 15 above)
Comment (linzichao) [reply Library/Homebrew/test/cask/audit_spec.rb:918]: Remove it.

File: Library/Homebrew/cask/audit.rb (hunk 16 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb:383]: It doesn't work. I still reserve it.

**Comment by linzichao** [comment]
@vitorgalvao @reitermarkus 
I have stuck in the `brew audit --skip-style` during GitHub CI. 
I know those failed cases result from my PR. What should I do to handle the huge number of failures?

**Comment by reitermarkus** [comment]
You can make it a warning instead of an error for now. Then we can switch over all casks to using it and make it an error once that's done.

**Comment by linzichao** [comment]
@vitorgalvao @reitermarkus
Please take a look.

File: Library/Homebrew/cask/audit.rb [hunk 17]
```diff
@@ -345,6 +348,75 @@ def bad_osdn_url?
...
+    end
+
+    def check_unnecessary_verified
+      return unless verified_present?
+      return unless url_match_homepage?
+      return unless verified_matches_url?
+
+      add_warning "The URL's #{domain} matches the homepage #{homepage}, " \
+                          "the `verified` parameter of the `url` stanza is unnecessary. " \
+                                  "See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-verified"
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:400]: Please fix the indentation here (and below) so that all these `"` line up.

File: Library/Homebrew/sorbet/files.yaml [hunk 18]
```diff
@@ -0,0 +1,962 @@
+false:
```
Comment (reitermarkus) [inline Library/Homebrew/sorbet/files.yaml:1]: Remove this file.

File: Library/Homebrew/test/support/fixtures/cask/Casks/version-latest.rb [hunk 19]
```diff
@@ -2,7 +2,7 @@
   version :latest
   sha256 :no_check
 
-  url "file://#{TEST_FIXTURE_DIR}/cask/caffeines.zip"
+  url "file://#{TEST_FIXTURE_DIR}/cask/caffeines.zip", verified: TEST_FIXTURE_DIR.to_s
```
Comment (reitermarkus) [inline Library/Homebrew/test/support/fixtures/cask/Casks/version-latest.rb:5]: These should not need to be changed. `file://` URLs should be excluded from this check.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Library/Homebrew/cask/audit.rb (hunk 17 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: Done.

File: Library/Homebrew/sorbet/files.yaml (hunk 18 above)
Comment (linzichao) [reply Library/Homebrew/sorbet/files.yaml]: Done.

File: Library/Homebrew/test/support/fixtures/cask/Casks/version-latest.rb (hunk 19 above)
Comment (linzichao) [reply Library/Homebrew/test/support/fixtures/cask/Casks/version-latest.rb]: Add `url_includes_file` to check whether URLs include `file://` prefix, and act as a flag to skip the `verified` examination.

File: Library/Homebrew/cask/audit.rb [hunk 20]
```diff
@@ -345,6 +348,80 @@ def bad_osdn_url?
...
+    def verified_matches_url?
+      strip_url_scheme(cask.url.to_s).include?(url_from_verified)
+    end
+
+    def verified_present?
+      cask.url.verified.present?
+    end
+
+    def url_includes_file?
+      cask.url.to_s.include?("file://")
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:394]:
```suggestion
      cask.url.to_s.start_with?("file://")
```

File: Library/Homebrew/cask/audit.rb (hunk 16 above)
Comment (reitermarkus) [reply Library/Homebrew/cask/audit.rb:386]: I think this should be `start_with?` then instead of `include?`.

File: Library/Homebrew/cask/url.rb [hunk 21]
```diff
@@ -7,8 +7,16 @@
 class URL
   extend T::Sig
 
+  ATTRIBUTES = [
+    :verified, :using,
+    :tag, :branch, :revisions, :revision,
+    :trust_cert, :cookies, :referer, :user_agent,
+    :data
+  ].freeze
+  private_constant :ATTRIBUTES
```
Comment (reitermarkus) [inline Library/Homebrew/cask/url.rb:16]: These are not used anymore remove them.

File: Library/Homebrew/sorbet/files.yaml [hunk 22]
```diff
@@ -0,0 +1,961 @@
+  - ./PATH.rb
```
Comment (reitermarkus) [inline Library/Homebrew/sorbet/files.yaml:1]: Remove this whole file.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Library/Homebrew/cask/url.rb (hunk 21 above)
Comment (linzichao) [reply Library/Homebrew/cask/url.rb]: OK.

File: Library/Homebrew/cask/audit.rb (hunk 20 above)
Comment (linzichao) [reply Library/Homebrew/cask/audit.rb]: Done.

File: Library/Homebrew/sorbet/files.yaml (hunk 22 above)
Comment (linzichao) [reply Library/Homebrew/sorbet/files.yaml]: Done.

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks, @linzichao, great work!

**Comment by n-h** [comment]
Hey all, I installed homebrew on OSX Catalina, about a week ago, and this morning when trying to install some casks, I started getting errors. Idk if this error is somehow related to this PR, but wanted to post, in case it is.

Lots of casks, showing `unknown keyword: verified`

```
$ brew --version
Homebrew 2.5.12
Homebrew/homebrew-core (git revision 0b7654; last commit 2020-11-28)

$ brew cask doctor
==> Tapping homebrew/cask
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 575225 (delta 1), reused 2 (delta 0), pack-reused 575215
Receiving objects: 100% (575225/575225), 240.94 MiB | 3.00 MiB/s, done.
Resolving deltas: 100% (409314/409314), done.
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/privatus.rb
Cask 'privatus' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/zeitgeist.rb
Cask 'zeitgeist' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/anka-build-cloud-registry.rb
Cask 'anka-build-cloud-registry' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/packetsender.rb
Cask 'packetsender' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/shotcut.rb
Cask 'shotcut' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/pdf-expert.rb
Cask 'pdf-expert' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/knockknock.rb
Cask 'knockknock' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/0-ad.rb
Cask '0-ad' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/daedalus-mainnet.rb
Cask 'daedalus-mainnet' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/recordit.rb
Cask 'recordit' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/spyder.rb
Cask 'spyder' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/videofusion.rb
Cask 'videofusion' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/runjs.rb
Cask 'runjs' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/stashcat.rb
Cask 'stashcat' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/stand.rb
Cask 'stand' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/gqrx.rb
Cask 'gqrx' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/oversight.rb
Cask 'oversight' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/anka-build-cloud-controller-and-registry.rb
Cask 'anka-build-cloud-controller-and-registry' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Invalid cask: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/anka-virtualization.rb
Cask 'anka-virtualization' definition is invalid: 'url' stanza failed with: unknown keyword: verified
Error: Cannot tap homebrew/cask: invalid syntax in tap!

$
```

**Comment by n-h** [comment]
Running `brew update`, made `brew cask list` and `brew cask doctor` successfully clone and setup the cask repo (it was the first time successfully doing that).

So `brew update`, looks to be a work-a-round, but it seems there may have been a breaking change somewhere.

**Comment by reitermarkus** [comment]
@n-h, your output shows version 2.5.12, this was released as part of Homebrew 2.6.0, so `brew update` should fix this.
